core/src/test/scala/unit/kafka/server/KafkaApisTest.scala

```diff
@@ -10100,6 +10110,31 @@ class KafkaApisTest extends Logging {
     verifyNoThrottling[MetadataResponse](requestChannelRequest)
   }
 
+  private def testConsumerListOffsetWithUnsupportedVersion(timestamp: Long, version: Short): Unit = {
+    val tp = new TopicPartition("foo", 0)
+    val targetTimes = List(new ListOffsetsTopic()
+      .setName(tp.topic)
+      .setPartitions(List(new ListOffsetsPartition()
+        .setPartitionIndex(tp.partition)
+        .setTimestamp(timestamp)).asJava)).asJava
+
+    val listOffsetRequest = ListOffsetsRequest.Builder
+      .forConsumer(false, IsolationLevel.READ_UNCOMMITTED, false, false)
+      .setTargetTimes(targetTimes).build(version)
```

Comment:
With https://github.com/apache/kafka/pull/16876, we can't build an incorrect request like that, right?

Reply:
Thanks for catching this. I should be more careful about it. Updated it.